perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -588,14 +590,28 @@ public Member getMemberByUserExtSource(PerunSession sess, Vo vo, UserExtSource u
 	}
 
 	public Member getMemberByUserExtSources(PerunSession sess, Vo vo, List<UserExtSource> ueas) throws InternalErrorException, MemberNotExistsException {
+		if (ueas == null) {
+			throw new InternalErrorException("Given userExtSources are null.");
+		}
+		if (ueas.isEmpty()) {
+			throw new InternalErrorException("Given userExtSources are empty.");
+		}
+
+		Set<Member> foundMembers = new HashSet<>();
+
 		for (UserExtSource ues: ueas) {
 			try {
-				return getMembersManagerImpl().getMemberByUserExtSource(sess, vo, ues);
+				foundMembers.add(getMembersManagerImpl().getMemberByUserExtSource(sess, vo, ues));
 			} catch (MemberNotExistsException e) {
-				// Ignore
+				throw new MemberNotExistsException("Member with userExtSources " + ueas + " doesn't exists.");
```

Comment:
I think you should ignore this exception, since you get list of UES, but only some of them might be registered in Perun, meaning you might find user only for some of them a those new UES you will add to the user.

Then you should throw ``MemberNotExistsException``, if ``foundMembers.size() == 0``.

Reply:
Ok, done.